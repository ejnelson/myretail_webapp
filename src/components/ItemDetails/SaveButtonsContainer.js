import React, { Component } from 'react';
import SaveButton from '../Shared/SaveButton';
import './ItemDetails.css';

export default class SaveButtonsContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  addToRegistry = () => console.log('addToRegistry');
  addToList = () => console.log('addToList');
  share = () => console.log('share');

  render() {
    return (
      <div className="saveButtonsContainer">
        <SaveButton
          action={() => this.addToRegistry()}
          buttonText="ADD TO REGISTRY"
        />
        <SaveButton action={() => this.addToList()} buttonText="ADD TO LIST" />
        <SaveButton action={() => this.save()} buttonText="SHARE" />
      </div>
    );
  }
}
