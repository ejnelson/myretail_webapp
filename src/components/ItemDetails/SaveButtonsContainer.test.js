import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SaveButtonsContainer from './SaveButtonsContainer';

describe('SaveButtonsContainer', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SaveButtonsContainer />, div);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<SaveButtonsContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('performs actions', () => {
    // Render the SaveButtonsContainer
    const container = shallow(<SaveButtonsContainer />);
    const addToRegistrySpy = jest.spyOn(container.instance(), 'addToRegistry');
    container.instance().addToRegistry();
    expect(addToRegistrySpy).toHaveBeenCalled();
    const addToListSpy = jest.spyOn(container.instance(), 'addToList');
    container.instance().addToList();
    expect(addToListSpy).toHaveBeenCalled();
    const shareSpy = jest.spyOn(container.instance(), 'share');
    container.instance().share();
    expect(shareSpy).toHaveBeenCalled();
  });
});
