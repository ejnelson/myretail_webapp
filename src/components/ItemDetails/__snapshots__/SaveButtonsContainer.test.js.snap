// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButtonsContainer renders correctly 1`] = `
<div
  className="saveButtonsContainer"
>
  <button
    className="saveButton"
    onClick={[Function]}
  >
    ADD TO REGISTRY
  </button>
  <button
    className="saveButton"
    onClick={[Function]}
  >
    ADD TO LIST
  </button>
  <button
    className="saveButton"
    onClick={[Function]}
  >
    SHARE
  </button>
</div>
`;
